packaging/build_rpm.py

```diff
@@ -0,0 +1,375 @@
+            "find /var/lib/mock/epel-7-x86_64/result "
+            "-name '*.noarch.rpm' -o -name '*.x86_64.rpm'",
+            _func=check_output,
+            ).strip()
+        check_call(
+                ['scp', '-F', SSH_CONFIG_FILE,
+                 'builder:{rpm}'.format(rpm=final_rpm),
+                 '.'])
+
+        # Power down the vagrant box
+        check_call(['vagrant', 'halt', 'builder'])
+
+    else:
+        raise RuntimeError('need either mock or vagrant to build')
+
+
+def build(source, spec_file_name):
+    """Builds the RPM"""
+    spec_file = path_join(source, 'packaging', spec_file_name)
+    check_mock_config()
+    # Get build options
+    rpmbuild_defines = get_rpmbuild_defines()
+    logger.info('defines', rpmbuild_defines)
+    rendered_spec_file = render_spec_file(spec_file, rpmbuild_defines)
+
+    # Build .src.rpm
+    check_call(
+            [MOCK, '--verbose', '--buildsrpm',
+             '--spec', rendered_spec_file,
+             '--sources', source,
+             ]
+            )
+    # Extract the .src.rpm file name.
+    for f in listdir(RESULT_DIR):
+        if f.endswith('.src.rpm'):
+            src_rpm = path_join(RESULT_DIR, f)
+            break
+    else:
+        raise RuntimeError('src rpm not found')
+
+    # Install our internal dependencies
+    install_dependencies(spec_file, source)
+
+    # mock strongly assumes that root is not required for building RPMs.
+    # Here we work around that assumption by changing the onwership of /opt
+    # inside the CHROOT to the mockbuild user
+    check_call([
+            MOCK, '--verbose', '--chroot', '--',
+            'chown', '-R', 'mockbuild', '/opt'
+            ])
+    # Build the RPM
+    check_call([
+            MOCK, src_rpm,
+            '--no-clean',
+            ])
+
+    # Extract the final .rpm file name.
+    for f in listdir(RESULT_DIR):
+        if f.endswith('.x86_64.rpm') or f.endswith('.noarch.rpm'):
+            final_rpm = path_join(RESULT_DIR, f)
+            break
+    else:
+        raise RuntimeError('final rpm not found')
+
+    return final_rpm
+
+
+def get_rpmbuild_defines():
+    """
+    Return a dictionary of macros from `version_info`
+    """
+    defines = {}
+    with open(path_join(SCRIPT_DIR, 'version_info')) as f:
+        for line in f:
+            line = line.strip()
+            if line and not line.startswith('#'):
+                k, v = line.split(' ', 1)
+                defines[k] = v
+
+    return defines
+
+
+def render_spec_file(spec_file, defines):
+    """
+    Apply the defines to the spec file. We don't use `--define` because
+    `mock`'s `--rpmbuild-opts` doesn't apply to `--buildsrpm`
+
+    Returns the path to the rendered spec file
+    """
+    with open(spec_file) as f:
+        spec_content = f.read()
+
+    for k, v in defines.items():
+        spec_content = spec_content.replace('%{' + k + '}', v)
+
+    spec_file_out = spec_file.replace('.spec', '.rendered.spec')
```

Comment:
let's make this a tempfile, because writing next to the .spec just clutters `git status`, and (so far) it's never useful to examine the rendered spec anyway

Reply:
I'm not super keen on hiding the rendered file in a tmpdir somewhere.
Added an entry to .gitignore. Shout if you object